### PR TITLE
datastore export

### LINKED ISSUE #1192: Export saved session to CSV
Add the ability to export a saved session to CSV, so it can be imported elsewhere. 

Likely via the session management page.

### PR DESCRIPTION
**Pull request by brentpicasso**
Allows the export of session to a CSV file, keeping to the original format as closely as possible.

resolves #1192 

Note, for this prelimiary PR you'll need to let the app generate a fresh DB - migration / schema management is WIP.

### CODE REVIEW

File: resource/datastore/migrations/2_migration.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+         smoothing INTEGER NOT NULL,
+         session_id INTEGER NOT NULL,
+         sample_rate INTEGER NOT NULL);
+
+INSERT INTO channel_migrate(name, units, min_value, max_value, smoothing, session_id, sample_rate) 
+SELECT c.name, c.units, c.min_value, c.max_value, c.smoothing, s.id, 25 FROM channel c, session s;
+
+DROP TABLE channel;
+
+ALTER TABLE channel_migrate RENAME TO channel;
```
Comment (ryandoherty) [inline resource/datastore/migrations/2_migration.sql]: Are all these migrations run in transactions? If not they need to be so if they fail the database isn't in a bad state.

**Comment by ryandoherty** [comment]
UI bug: there are two progress bars in the export popup (I think) http://i.imgur.com/yURqXPF.png

**Comment by brentpicasso** [comment]
oh, that's the horizontal rule that's at the top of every dialog. I can add some padding.

File: autosportlabs/racecapture/datastore/datastore.py [hunk 2]
```diff
@@ -351,44 +393,26 @@ def get_channel(self, name):
...
+                        Column('date2', Integer, nullable=False))
 
-        self._conn.execute("""CREATE TABLE datalog_channel_map
-        (datalog_id INTEGER NOT NULL, channel_id INTEGER NOT NULL)""")
-
-        self._conn.execute("""CREATE TABLE datalog_event_map
-        (datalog_id INTEGER NOT NULL, event_id INTEGER NOT NULL)""")
-
-        self._conn.commit()
+        print('create all')
```
Comment (brentpicasso) [inline autosportlabs/racecapture/datastore/datastore.py]: will fix

Comment (brentpicasso) [inline autosportlabs/racecapture/datastore/datastore.py]: fixed

File: autosportlabs/racecapture/datastore/datastore.py [hunk 3]
```diff
@@ -278,66 +280,106 @@ def __str__(self):
...
+            if len(current_index):
+                channels.insert(0, channels.pop(current_index[0]))
+
         c = self._conn.cursor()
 
-        self._channels = []
-        c.execute("""SELECT name, units, min_value, max_value, smoothing
-        from channel""")
+        channels = []
+        where = '' if session_id is None else 'WHERE session_id = {}'.format(session_id)
```
Comment (brentpicasso) [inline autosportlabs/racecapture/datastore/datastore.py]: fixed.

File: resource/datastore/migrations/2_migration.sql (hunk 1 above)
Comment (brentpicasso) [inline resource/datastore/migrations/2_migration.sql]: fixed so migration runs in a transaction.

File: autosportlabs/racecapture/views/analysis/analysisview.py [hunk 4]
```diff
@@ -271,8 +282,72 @@ def on_add_session(self, instance, session):
...
+    
+    def on_export_session(self, instance, session):
+
+        def _export_session(instance):
+
+            def _do_export_session(filename):
+
+                _do_export_session.cancelled = False
+
+                def _progress_cb(pct):
```
Comment (ryandoherty) [inline autosportlabs/racecapture/views/analysis/analysisview.py]: Ok ¯\_(ツ)_/¯

**Comment by ryandoherty** [comment]
:shipit: !

**Pull request merged**